### ISSUE TITLE
Accordion component suggested optimizations

### ISSUE THREAD

**Issue by aaronhans** [cagov/design-system#400]
We have some great suggestions to improve the accordion component from state employees, discussion here https://github.com/Office-of-Digital-Innovation/California-State-Template/issues/184

- Would be cool to evaluate the value of the animation. I like it personally and we have done some user testing on this over a year ago but if the animation isn't adding much value we could consider removing it to simplify the code.
- There is an opportunity to simplify code and improve progressive enhancement based on that discussion while retaining current functionality

**Comment by kkroon** [comment]
Thank you for creating this, @aaronhans
(I looked at an issue template I thought I'd be obliged to use and thought "ain't nobody got time fo' dat ...")

**Comment by aaronhans** [comment]
Doing some prototyping here https://codepen.io/aaronhans/pen/ZEaEagR it looks like this is going to be a massive improvement. Thanks @kkroon!

Will be experimenting with ways to handle the svgs, I like @kkoryaka's suggestion to create the shapes in pure CSS so they can accept CSS variables fill colors.

**Comment by rkojik** [comment]
> Will be experimenting with ways to handle the svgs, I like @kkoryaka's suggestion to create the shapes in pure CSS so they can accept CSS variables fill colors.

We could set the fill of an SVG with CSS as well, and provide an accessible label via <text> tag within the SVG which can communicate the state (the goal of the icon) of expanded or collapsed. Though, I imagine we're already communicating that through other semantics, so the icon may be visual: in which case, yeah, CSS shape is better.

**Comment by kkroon** [comment]
@rkojik Since the details element is HTML native, I suspect it already has a mechanism to indicate its state (e.g., the _open_ attribute). I am willing to pass it by a screen reader ...

**Comment by aaronhans** [comment]
I have updated the codepen to use a plus/minus indicator composed of a single div with box-shadows using the CSS variable as the background color value. This is inserted into the summary element in the connectedCallback with aria-hidden="true"

https://codepen.io/aaronhans/pen/ZEaEagR

**Comment by jaredwyee** [comment]
@chachasikes @aaronhans not entirely sure how to do a final QA on this. I don't appreciate what I'm really looking for on the component. 

Changelog wise, seems a bit odd that the new version 2.0.X is the upgrade but under Previous versions there's reference to 2.0.11 and lower numbers
![image](https://user-images.githubusercontent.com/85529382/157732191-30ea4a5f-9853-44f1-80e9-0c10735948b6.png)

**Comment by chachasikes** [comment]
@jaredwyee I see what you mean about previous versions - you have to really read it to understand what's going on, which is the opposite of how we try to give essential meaning through headers.
* Maybe make them not bold & BOLD the @cagov/accordion.
* Change "Previous versions" to "Deprecated npm packages" --- I know I have seen similar notes in other CHANGELOGs - curious to look for other examples.

**Comment by chachasikes** [comment]
@jaredwyee To QA - making sure the accordion behavior is as expected would be good - and that it conforms to what we describe in the docs.

Also this is the first time we are QA'ing a version bump - so it's worth it to talk it through for next time.

**Comment by jaredwyee** [comment]
@chachasikes thanks, as a user interactive via desktop browser, the accordion functions as I expect. Seems like a big change in the component code is less reliance on JS to improve progressive enhancement. That doesn't seem easily testable for me?

**Comment by chachasikes** [comment]
Correct, the change is mostly in the code & we've reviewed that & you can't test it - but verifying it works as expected, given a significant change, it enough for now. Thank you.

**Comment by aaronhans** [comment]
Thanks @jaredwyee! We have changed all the underlying code and radically simplified it but there are no expected changes in the normal UI. So if it is working as usual we are in good shape.
